src/rez/tests/test_packages_order.py

```diff
@@ -289,6 +291,287 @@ def test_pod(self):
+        self._test_reorder(CustomPackageOrder({
+            "pypa": ['1+<1.1', "2", "1.1+<2"],
+        },
+            version_orderer=PyPAPackageOrder(prerelease="b"),
+        ), "pypa", [
+            # Release and prerelease up to beta for 1+<1.1 to the front
+            '1.0.2.rc2',
+            '1.0.2.rc1',
+            '1.0.2.b1',
+            '1.0.1',
+            '1.0.1.rc2',
+            '1.0.1.rc1',
+            '1.0.1.b2',
+            '1.0.1.b1',
+            '1.0.0.post1',
+            '1.0.0+local',
+            '1.0.0',
+            '1.0.0.rc2',
+            '1.0.0.rc1',
+            '1.0.0.b1',
+            # Followed by 1+<1.1 prerelease below beta
+            '1.0.2.a1',
+            '1.0.1.a1',
+            '1.0.0.a2',
+            '1.0.0.a1',
+            # Followed by version 2
+            '2.0.0.a2',
+            '2.0.0.a1',
+            # Lastly followed by 1.1
+            '1.1.0.b2',
+            '1.1.0.b1',
+            '1.1.0.a1',
+        ])
+
+    def test_repr(self):
+        """Validate we can represent a CustomPackageOrder as a string."""
+        self.assertEqual(
+            "CustomPackageOrder({"
+            "'python': [VersionRange('1.2'), VersionRange('5.6')], "
+            "'pymum': [VersionRange('2'), VersionRange('1')]})",
+            repr(CustomPackageOrder({
+                "python": ["1.2", "5.6"],
+                "pymum": ["2", "1"],
+            })))
+
+    def test_pod(self):
+        """Validate we can save and load a CustomPackageOrder to its pod representation."""
+        self._test_pod(CustomPackageOrder({
+            "python": ["1.2", "5.6", "3.4"],
+            "pymum": ["2", "1", "3"],
+        }))
+
+
+class TestPyPAPackageOrder(_BaseTestPackagesOrder):
+    """Test case for the PyPAPackageOrder class"""
+
+    def test_reorder(self):
+        """Validate package ordering with a PyPAPackageOrder"""
+        # First test that we can sort packages prerelease=None,
+        # so we expect non-prerelease versions on top.
+        orderer = PyPAPackageOrder()
+        expected = [
+            # Release and post releases first.
+            '1.0.1',
+            '1.0.0.post1',
+            '1.0.0+local',
+            '1.0.0',
+
+            # Followed by higher version prerelease versions, sorted according to pypa
+            '2.0.0.a2',
+            '2.0.0.a1',
+            '1.1.0.b2',
+            '1.1.0.b1',
+            '1.1.0.a1',
+            '1.0.2.rc2',
+            '1.0.2.rc1',
+            '1.0.2.b1',
+            '1.0.2.a1',
+            '1.0.1.rc2',
+            '1.0.1.rc1',
+            '1.0.1.b2',
+            '1.0.1.b1',
+            '1.0.1.a1',
+            '1.0.0.rc2',
+            '1.0.0.rc1',
+            '1.0.0.b1',
+            '1.0.0.a2',
+            '1.0.0.a1',
+        ]
+        self._test_reorder(orderer, "pypa", expected)
+
+        # Test that we can allow RC versions ahead of release
+        orderer = PyPAPackageOrder(prerelease="rc")
```

Comment:
It is a risk allowance setting.
Consider 2.0.a1, 1.3.rc1, 1.2.b1, and 1.1.
Typically users do not want the alpha version of some software.  Most users want the release version.  This would not *prevent* a user from using 1.1, but would allow them to opt into it.

PyPA sorting is almost a two dimensional version scheme, where you may want to sort the release versions as preferred or you may want the alpha versions as preferred.

The idea here is to mirror how pip works, where you need to specifically ask for an alpha version.  If you just do `pip install requests` you will get the latest release version.